include/jemalloc/internal/emap.h

```diff
@@ -234,15 +234,34 @@ emap_edata_lookup_batch(tsd_t *tsd, emap_t *emap, size_t nptrs,
 
 	/* Avoids null-checking tsdn in the loop below. */
 	util_assume(tsd != NULL);
+	rtree_ctx_t *rtree_ctx = tsd_rtree_ctxp_get(tsd);
 
 	for (size_t i = 0; i < nptrs; i++) {
-		emap_full_alloc_ctx_t full_alloc_ctx;
 		const void *ptr = ptr_getter(ptr_getter_ctx, i);
+		/*
+		 * Reuse the edatas array as a temp buffer, lying a little about
+		 * the types.
+		 */
+		r_edatas[i] = (edata_t *)rtree_leaf_elm_lookup(tsd_tsdn(tsd),
```

Comment:
I ended up making it more complex :(. I had convinced myself that this sort of cheating was allowed, but it's not really without a union. Given that we do actually care about the ordering of the reuse, and this could be broken by TBAA, made it UB-free.